src/platforms/outreach/actions/prospects/update.ts

```diff
@@ -10,35 +33,23 @@ export default action(
     mutation: true,
     schema: z.object({
       id: z.number(),
-      attributes: z
-        .object({
-          firstName: z.string().nullish(),
-          lastName: z.string().nullish(),
-          occupation: z.string().nullish(),
-          addressCity: z.string().nullish(),
-          addressCountry: z.string().nullish(),
-          addressState: z.string().nullish(),
-          addressStreet: z.string().nullish(),
-          addressStreet2: z.string().nullish(),
-          addressZip: z.string().nullish(),
-        })
-        .catchall(
-          z
-            .string()
-            .regex(
-              /^custom[0-9]+$/,
-              'Custom fields must be of the form custom[0-9]+',
-            )
-            .transform((key) => key as `custom${number}`)
-            .nullable(),
-        ),
+      attributes: z.preprocess((input) => {
+        const attributes = attributesSchema.parse(input);
+        const customFields = customFieldsSchema.parse(
```

Comment:
Any specific reason for not using $native?

Reply:
or have a field called `customFields` where they stick it in?

Reply:
this `customFields` thing is only used internally to group the given custom fields, the customer won't see the field